docs/src/test/scala/docs/http/scaladsl/server/PlayRoutesComparisonSpec.scala

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (C) 2009-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package docs.http.scaladsl.server
+
+import akka.http.scaladsl.common.ToNameReceptacleEnhancements
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server._
+import akka.http.scaladsl.testkit.ScalatestRouteTest
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+
+class PlayRoutesComparisonSpec extends AnyWordSpec with Matchers with ScalatestRouteTest {
+
+  object Clients {
+    def list(): String = "clientA,clientB,clientC"
+
+    def get(id: Long): String = "clientB"
+  }
+
+  "Play examples" should {
+
+    "show a fixed URL" in {
+      val clientsAll: Route =
+        // #fixed
+        (get & path("clients" / "all")) {
+          complete(Clients.list())
+        }
+      // #fixed
+
+      // #fixed-test
+      Get("/clients/all") ~> clientsAll ~> check {
+        responseAs[String] shouldEqual "clientA,clientB,clientC"
+      }
+      // #fixed-test
+    }
+
+    "show reading a long" in {
+      val clientById: Route =
+        // #long
+        (get & pathPrefix("client" / LongNumber)) { id =>
+          complete(Clients.get(id))
+        }
+      // #long
+
+      // #long-test
+      Get("/client/321433") ~> clientById ~> check {
+        responseAs[String] shouldEqual "clientB"
+      }
+      // #long-test
+    }
+
+    "use multiple segments" in {
+      def download(name: String) = s"$name: file contents"
+
+      val files: Route =
+        // #segments
+        (get & pathPrefix("files" / Segments)) { names =>
+          val name = names.mkString("/")
+          complete(download(name))
+        }
+      // #segments
+
+      // #segments-test
+      Get("/files/images/logo.png") ~> files ~> check {
+        responseAs[String] shouldEqual "images/logo.png: file contents"
+      }
+      // #segments-test
+    }
+
+    "require a parameter" in {
+      def getPage(name: String) = s"The requested [$name]."
+
+      val pageParameter: Route =
+        // #mandatory-parameter
+        (get & parameter("page")) { page =>
+          complete(getPage(page))
+        }
+      // #mandatory-parameter
+
+      // #mandatory-parameter-test
+      Get("/?page=example.txt") ~> pageParameter ~> check {
+        responseAs[String] shouldEqual "The requested [example.txt]."
+      }
+      // #mandatory-parameter-test
+    }
+
+    "accept a parameter" in {
+      def listAll(version: Option[String]) = version.fold("ff")(_ => "aa,bb,cc")
+
+      val optionalPageParameter: Route =
+        // #optional-parameter
+        get {
+          path("api" / "list-all")(parameter("version".?) { version =>
```

Comment:
Let's either use proper nesting with one directive per line, or use `&`. I.e.

```suggestion
          (path("api" / "list-all") & parameter("version".?)) { version =>
```